### ISSUE TITLE
Add scrollbar for first level submenus

### ISSUE THREAD

**Issue by joomlagate** [mortalis13/mod_menu_adv#3]
hi, at first thank you for this great module.

Today I tested it, and found out that it will add scrollbar for second or third level submenus, such as "extensions -> module manager -> site -> many modules show here ".

But , for the first level submenu, it will NOT add scrollbar. For example, if I install more than 30 components on the same Joomla website, yes , more than 30, then the backend "Components" menu will have a very long first level submenu, which will exceed the screen bottom border. Since the Control Panel page of Joomla backend is very short, it does not have a scrollbar, and the "Components" menu does not have a scrollbar either, so , I can not see the whole submenu of "Components". 

I think you have understand my issue now. Could you please add a new feature of "add scrollbar for first level submenu when necessary" to this module?

Thank you.

**Comment by mortalis13** [comment]
Hi.
Thank you for your notice.
I'll try to work with it, but cannot promise very fast solution for now.

**Comment by mortalis13** [comment]
Sorry I couldn't answer your question immediately. I haven't work with this extension for a long time and needed some time to remember how it works.

So for now it's impossible to realize unfortunately. And I cannot invent anything.
When I created the module I tried to provide the scrolling for the second level submenus when they have the third sublevel. But when I set the **CSS** `overflow-y` and `max-height` properties the next level won't show, it's enclosed within its parent's level with its size limits.

The same occurs with the 1st-2d levels relations.

And as I tested the Joomla default admin menu, if you have many components in the appropriate submenu they will not scroll either.
So the Joomla is limited in this question too.

And to solve it I guess the menu will need some other way to show a lot of items in a limited space then just setting simple CSS properties.

---

## Technical part

This is the menu with duplicated items (to fill the space) and `overflow-y: auto` and `max-height: 500px`:
![](http://s4.postimg.org/bhoqfv8ml/image.png)

When the mouse is over an item with submenu, the latter is hidden within the common box:
![](http://s4.postimg.org/hl6azrywd/image.png)

If you drag and release a little the Contacts item without pressing it (just to focus the item) and then press the `Right` key a couple of times you'll see the hidden submenu:
![](http://s4.postimg.org/5i0zc7nu5/image.png)

This wasn't solved by changing the width of the 1st level menu box (the hidden part is still considered to "overflow" the space):
![](http://s4.postimg.org/dkz5nj8fh/image.png)

**Comment by joomlagate** [comment]
Yes, I know Joomla core itself can not add scrollbar for too many items on the "Components" menu, so I hope there is some 3rd party extension can solve this.

Maybe you can submit a patch to Joomla core team to solve this?

Thank you.
